devicemodel/hw/pci/virtio/virtio_net.c

```diff
@@ -896,17 +901,32 @@ virtio_net_init(struct vmctx *ctx, struct pci_vdev *dev, char *opts)
 	 */
 	net->tapfd = -1;
 
-	if (!devname) {
-		WPRINTF(("virtio_net: devname NULL\n"));
+	if (!devopts) {
+		WPRINTF(("virtio_net: invalid optional argument\n"));
 		free(net);
 		return -1;
 	}
 
-	if ((strstr(devname, "tap") != NULL) ||
-	    (strncmp(devname, "vmnet", 5) == 0))
-		virtio_net_tap_setup(net, devname);
+	vtopts = tmp = strdup(opts);
+	if ((strstr(tmp, "tap") != NULL)
+			|| (strncmp(tmp, "vmnet",5) == 0)) {
+		type = strsep(&tmp, "=");
+		name = strsep(&tmp, ",");
+	}
+
+	if (strncmp(tmp, "mac_seed",8) == 0) {
+		strsep(&tmp, "=");
+		mac_seed = tmp;
+	}
 
-	free(devname);
+	if ((type != NULL) && (name != NULL)) {
+		snprintf(fname, IFNAMSIZ, "%s_%s", type, name);
```

Comment:
I did not pay attention to that part... we should just use the name given by the user as the TAP device name. We need to fix this.

Reply:
> I did not pay attention to that part... we should just use the name given by the user as the TAP device name. We need to fix this.

As we talk, we could change this logic.

BTW, if some existing device name tap_TaaG, use have to set "tap=tap_TaaG".